app/models/sponsorship.rb

```diff
@@ -2,6 +2,7 @@ class Sponsorship < ApplicationRecord
   LEVELS = %w[gold silver bronze tag media devrel].freeze
   METAL_LEVELS = %w[gold silver bronze].freeze
   STATUSES = %w[none pending live].freeze
+  SPONSORABLE_TYPES = [Tag, ActsAsTaggableOn::Tag].map(&:name).freeze
```

Comment:
Should we simplify this a bit?

```suggestion
  SPONSORABLE_TYPES = %w[Tag ActsAsTaggableOn::Tag].freeze
```

The only case where doesn't return the string representation of the class name is anonymous modules, which return `nil`:

```ruby
Module.new.name
#=> nil
```

Reply:
The difference I think between the two versions (names mapped from array of class constants, and array of strings that look like class constants) is that the `%w` array doesn't require the constants to exist - if we either added new classes to this list of sponsorables, or removed ActsAsTaggableOn from the code, it seems like the string could pass validation while the sponsorship would then be unloadable. I am actually trying to ensure the provided value is both on the allow list, and particularly is a something we could load as an associated model. While the original version called constantize on the input (and looked for no error), nothing covers that any more.

With that said, since we would be loading these by id as a polymorphic association, just ensuring it's a constant is too weak to prevent all errors, and validating the form input according to the allow list seems sufficient.

Reply:
> With that said, since we would be loading these by id as a polymorphic association, just ensuring it's a constant is too weak to prevent all errors

Yes, this is where I was coming from. I'm open to your version too, the rationale is sound but it doesn't seem to be a pattern we use for similar cases elsewhere.